src/app/common/PolicyTypeTag.vue

```diff
@@ -35,36 +34,30 @@ import TrafficPermissionIconUrl from '@/assets/images/policies/TrafficPermission
 import VirtualOutboundIconUrl from '@/assets/images/policies/VirtualOutbound.png'
-import { useStore } from '@/store/store'
 
-const store = useStore()
-
-type PolicyTagDefinition = { iconUrl: string | null }
-
-const POLICIES: Record<string, PolicyTagDefinition> = {
-  CircuitBreaker: { iconUrl: CircuitBreakerIconUrl },
-  FaultInjection: { iconUrl: FaultInjectionIconUrl },
-  HealthCheck: { iconUrl: HealthCheckIconUrl },
-  MeshAccessLog: { iconUrl: TrafficLogIconUrl }, // TODO: Update with new icon when/if available.
-  MeshCircuitBreaker: { iconUrl: CircuitBreakerIconUrl }, // TODO: Update with new icon when/if available.
-  MeshGateway: { iconUrl: null }, // TODO: Update with new icon when/if available.
-  MeshGatewayRoute: { iconUrl: null }, // TODO: Update with new icon when/if available.
-  MeshHealthCheck: { iconUrl: HealthCheckIconUrl }, // TODO: Update with new icon when/if available.
-  MeshProxyPatch: { iconUrl: ProxyTemplateIconUrl }, // TODO: Update with new icon when/if available.
-  MeshRateLimit: { iconUrl: RateLimitIconUrl }, // TODO: Update with new icon when/if available.
-  MeshRetry: { iconUrl: RetryIconUrl }, // TODO: Update with new icon when/if available.
-  MeshTimeout: { iconUrl: TimeoutIconUrl }, // TODO: Update with new icon when/if available.
-  MeshTrace: { iconUrl: TrafficTraceIconUrl }, // TODO: Update with new icon when/if available.
-  MeshTrafficPermission: { iconUrl: TrafficPermissionIconUrl }, // TODO: Update with new icon when/if available.
-  ProxyTemplate: { iconUrl: ProxyTemplateIconUrl },
-  RateLimit: { iconUrl: RateLimitIconUrl },
-  Retry: { iconUrl: RetryIconUrl },
-  Timeout: { iconUrl: TimeoutIconUrl },
-  TrafficLog: { iconUrl: TrafficLogIconUrl },
-  TrafficPermission: { iconUrl: TrafficPermissionIconUrl },
-  TrafficRoute: { iconUrl: TrafficRouteIconUrl },
-  TrafficTrace: { iconUrl: TrafficTraceIconUrl },
-  VirtualOutbound: { iconUrl: VirtualOutboundIconUrl },
+const POLICY_ICON: Record<string, string> = {
+  CircuitBreaker: CircuitBreakerIconUrl,
+  FaultInjection: FaultInjectionIconUrl,
+  HealthCheck: HealthCheckIconUrl,
+  MeshAccessLog: TrafficLogIconUrl, // TODO: Update with new icon when/if available.
+  MeshCircuitBreaker: CircuitBreakerIconUrl,
+  MeshFaultInjection: FaultInjectionIconUrl,
+  MeshHealthCheck: HealthCheckIconUrl, // TODO: Update with new icon when/if available.
+  MeshProxyPatch: ProxyTemplateIconUrl, // TODO: Update with new icon when/if available.
+  MeshRateLimit: RateLimitIconUrl,
+  MeshRetry: RetryIconUrl,
+  MeshTimeout: TimeoutIconUrl,
+  MeshTrace: TrafficTraceIconUrl,
+  MeshTrafficPermission: TrafficPermissionIconUrl,
+  ProxyTemplate: ProxyTemplateIconUrl,
+  RateLimit: RateLimitIconUrl,
+  Retry: RetryIconUrl,
+  Timeout: TimeoutIconUrl,
+  TrafficLog: TrafficLogIconUrl,
+  TrafficPermission: TrafficPermissionIconUrl,
+  TrafficRoute: TrafficRouteIconUrl,
+  TrafficTrace: TrafficTraceIconUrl,
+  VirtualOutbound: VirtualOutboundIconUrl,
 }
```

Comment:
In the future these icons, should  be the sort of thing we do in the layer between the API and the actual data (as introduced in https://github.com/kumahq/kuma-gui/pull/1242)?

Reply:
Possibly, yes. That could be attached to loading `/policies`, but the bigger question would be whether we could get a complete set of icons for the current set of policies as with the advent of new policy types especially, we’re missing quite a few.

Reply:
We'd likely have a generic icon for new and plugin types.